app/src/main/java/com/realityexpander/tasky/presentation/login_screen/LoginScreen.kt

```diff
@@ -57,113 +86,201 @@ fun LoginScreen(
     }
 
     BackHandler(true) {
-        navigateToRegister()
+        /* do nothing, should we ask the user to quit? */
+        // navigateToRegister()
     }
 
-    Box(
-        modifier = Modifier.fillMaxSize()
-    ) {
-        if(loginState.isLoading) {
-            Spacer(modifier = Modifier.height(8.dp))
-            CircularProgressIndicator(
-                modifier = Modifier.align(alignment = Alignment.Center)
-            )
+    // Check keyboard open/closed (how to make this a function?)
+    val view = LocalView.current
+    var isKeyboardOpen by remember { mutableStateOf(false) }
+    DisposableEffect(view) {
+        val listener = ViewTreeObserver.OnGlobalLayoutListener {
+            isKeyboardOpen = ViewCompat.getRootWindowInsets(view)
+                ?.isVisible(WindowInsetsCompat.Type.ime()) ?: true
+        }
+        view.viewTreeObserver.addOnGlobalLayoutListener(listener)
+
+        onDispose {
+            view.viewTreeObserver.removeOnGlobalLayoutListener(listener)
         }
     }
```

Comment:
In Compose we finally don't need this crap anymore, you can just do this (still marked as experimental, but didn't have issues with it yet):
```
val controller = LocalSoftwareKeyboardController.current
controller?.show()
controller?.hide()
```

Reply:
I understand that it is easy to open and close the keyboard with compose, but im changing the UI depending on if the keyboard is open or close. There is no easy way to detect if the keyboard is open. I looked quite a bit and found different solutions (most didnt work) and this is the one that worked all the time.

Reply:
Ah okay, thanks for explaining that makes sense